core/src/main/java/org/infinispan/configuration/cache/HashConfiguration.java

```diff
@@ -138,6 +148,10 @@ public float capacityFactor() {
       return capacityFactor.get();
    }
 
+   public KeyPartitionerFactory keyPartitionerFactory() {
+      return keyPartitionerFactory.get();
+   }
+
```

Comment:
But I agree that for other cases this may be dangerous...
Will check the component registry stratgey